code/src/hooks/use-add-to-cart-or-open-drawer/useAddToCartOrOpenDrawer.tsx

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from "react";
+
+import CartDrawer from "@/containers/cart-drawer/CartDrawer";
+
+import { useDrawerContext } from "@/context/drawer/DrawerContext";
+import useAddToCart from "@/hooks/use-add-to-cart/useAddToCart";
+import useGetCart from "@/hooks/use-get-cart/useGetCart";
+import { Product } from "@/types/product.types";
+
+type MinimalRequiredProduct = Omit<Product, "status">;
+
+const useAddToCartOrOpenDrawer = <T extends MinimalRequiredProduct>(
+  product: T
+) => {
+  const { data: cartData, isFetching: isCartFetching } = useGetCart();
```

Comment:
what heppened if it's loading or error ? maybe also handle it ?

Reply:
Handled loading state:
https://github.com/user-attachments/assets/a4c1d384-b3e6-414c-ab82-659721cf183a

As for error state, I think it it not required here because it is handled by another hook
https://github.com/user-attachments/assets/d6775e71-3595-4d75-8302-e191073c3748